src/devices/purifier/class.ts

```diff
@@ -0,0 +1,217 @@
+import { Device } from '../../device';
+import {
+	BinaryPropertyStatus,
+	PropertyStatus,
+	StatusCode,
+	valueBinary,
+	ValueSingle,
+	ValueType,
+	mode,
+} from './properties';
+
+export class Purifier extends Device {
+	constructor(props) {
+		super(props);
+	}
+
+	/**
+	 * Queues a power on action
+	 */
+	queuePowerOn(): void {
+		this.queuePropertyStatusUpdate({
+			valueSingle: {
+				code: ValueSingle.POWER_ON,
+			},
+			statusCode: StatusCode.POWER,
+			valueType: ValueType.SINGLE,
+		});
+	}
+
+	/**
+	 * Queues a power off action
+	 */
+	queuePowerOff(): void {
+		this.queuePropertyStatusUpdate({
+			valueSingle: {
+				code: ValueSingle.POWER_OFF,
+			},
+			statusCode: StatusCode.POWER,
+			valueType: ValueType.SINGLE,
+		});
+	}
+
+	/**
+	 * Convert mode string to binary.
+	 * @return     {string}  mode binary
+	 */
+	existMode = (mode: string): string => {
```

Comment:
existMode -> stringToMode or something like this would be better

Reply:
All right, thanks!

By the way, would you prefer the private function? I don't plan to use it from outside.

Reply:
sure, lets make it private

Reply:
ok! i'm fix it.